clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/view/EcsInstanceProvider.java

```diff
@@ -0,0 +1,171 @@
+package com.netflix.spinnaker.clouddriver.ecs.view;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.model.DescribeInstanceStatusRequest;
+import com.amazonaws.services.ec2.model.InstanceStatus;
+import com.amazonaws.services.ecs.AmazonECS;
+import com.amazonaws.services.ecs.model.ContainerInstance;
+import com.amazonaws.services.ecs.model.DescribeContainerInstancesRequest;
+import com.amazonaws.services.ecs.model.DescribeTasksRequest;
+import com.amazonaws.services.ecs.model.ListClustersRequest;
+import com.amazonaws.services.ecs.model.ListClustersResult;
+import com.amazonaws.services.ecs.model.Task;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.EcsCloudProvider;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsInstance;
+import com.netflix.spinnaker.clouddriver.model.InstanceProvider;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Component
+public class EcsInstanceProvider implements InstanceProvider<EcsInstance> {
+
+  private final String cloudProvider = EcsCloudProvider.ID;
+
+  @Autowired
+  private AccountCredentialsProvider accountCredentialsProvider;
+
+  @Autowired
+  private AmazonClientProvider amazonClientProvider;
+
+
+  @Override
+  public String getCloudProvider() {
+    return cloudProvider;
+  }
+
+  @Override
+  public EcsInstance getInstance(String account, String region, String id) {
+    if (!isValidId(id, region))
+      return null;
+
+    EcsInstance ecsInstance = null;
+
+    AWSCredentialsProvider awsCredentialsProvider = getCredentials(account).getCredentialsProvider();
+    AmazonECS amazonECS = amazonClientProvider.getAmazonECS(account, awsCredentialsProvider, region);
+    AmazonEC2 amazonEC2 = amazonClientProvider.getAmazonEC2(account, awsCredentialsProvider, region);
+
+    Task ecsTask = getTask(amazonECS, id);
+    InstanceStatus instanceStatus = getEC2InstanceStatus(amazonEC2, getContainerInstance(amazonECS, ecsTask));
+
+    if (ecsTask != null && instanceStatus != null) {
+      ecsInstance = new EcsInstance(id, ecsTask, instanceStatus);
+    }
+
+    return ecsInstance;
+  }
+
+  @Override
+  public String getConsoleOutput(String account, String region, String id) {
+    return null;
```

Comment:
Why was this not done?  Is it because it's not possible with tasks?

Reply:
correct